src/org/zaproxy/zap/extension/ascanrules/TestPathTraversal.java

```diff
@@ -380,48 +380,52 @@ public void scan(HttpMessage msg, String param, String value) {
                 String urlfilename = msg.getRequestHeader().getURI().getName();
                 String prefixedUrlfilename;
 
-                //for the url filename, try each of the prefixes in turn
-                for (String prefix : LOCAL_FILE_RELATIVE_PREFIXES) {
-
-                    prefixedUrlfilename = prefix + urlfilename;
-                    msg = getNewMsg();
-                    setParameter(msg, param, prefixedUrlfilename);
-
-                    //send the modified message (with the url filename), and see what we get back
-        			try {
-        	            sendAndReceive(msg);
-        			} catch (SocketException|IllegalStateException|UnknownHostException|IllegalArgumentException|InvalidRedirectLocationException|URIException ex) {
-        				if (log.isDebugEnabled()) log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
-        						" when accessing: " + msg.getRequestHeader().getURI().toString());
-        				continue; //Something went wrong, move to the next prefix in the loop
-        			}
-
-                    //did we get an Exception or an Error?
-                    errorMatcher = errorPattern.matcher(msg.getResponseBody().toString());
-                    if ((msg.getResponseHeader().getStatusCode() == HttpStatusCode.OK)
-                            && (!errorMatcher.find())) {
-
-                        //if it returns OK, and the random string above did NOT return ok, then raise an alert
-                        //since the filename has likely been picked up and used as a file name from the parameter
-                        bingo(
-                                Alert.RISK_HIGH,
-                                Alert.CONFIDENCE_MEDIUM,
-                                null,
-                                param,
-                                prefixedUrlfilename,
-                                null,
-                                msg);
-
-                        // All done. No need to look for vulnerabilities on subsequent parameters 
-                        // on the same request (to reduce performance impact)
-                        return;
-                    }
-                    // Check if the scan has been stopped
-                    // if yes dispose resources and exit
-                    if (isStop()) {
-                        // Dispose all resources
-                        // Exit the plugin
-                        return;
+                //url file name may be empty, i.e. there is no file name for next check
+                if (!urlfilename.isEmpty()) {

```

Comment:
This will lead to a `NullPointerException` if the URL has an empty path, it should be checked that's non-null too. Also, this condition should be moved to the above if (the one with `msg.getResponseHeader().getStatusCode() != HttpStatusCode.OK`), no need to log that we can check if we are not going to.